python/Ganga/Core/MonitoringComponent/Local_GangaMC_Service.py

```diff
@@ -101,12 +106,23 @@ def _execUpdateAction(self):
                 else:
                     action.callback_Failure()
 
+            self._currently_running_command = False
+
 # Create the thread pool
 
 
 def _makeThreadPool(threadPoolSize=THREAD_POOL_SIZE, daemonic=True):
+    global ThreadPool
+    if ThreadPool and len(ThreadPool) != 0:
+        #from Ganga.Core.exceptions import GangaException
+        #raise GangaException("Cannot doubbly init the ThreadPool! ThreadPool already populated with threads")
+        log.debug("Found a thread pool already in existance, wiping it and startig again!")
+        for i in ThreadPool:
+            del i
+        del ThreadPool[:]
+        ThreadPool = []
```

Comment:
This could be replaced if it was actually working in the disable and stop methods but as this ```del ThreadPool[:]``` is not working there we're left with a TreadPool object when starting up again.
I suspect this could be a case of the object being marked for deletion but still existing as ThreadPool is not set to ```[]``` at the end of the shutdown.